File: 3_1/p012.py
# n = int(input())
# i = 2
# res = 'Yes'
# while i**2 <= n:
#     if n % i != 0:
#         i += 1
#         continue
#     else:
#         res = 'No'
#         break
# print(res)

def prime(n):
    max_num = int(n**0.5)
    for i in range(2, max_num+1):
        if n % i == 0:
            return False
    return True

if __name__ == '__main__':
    n = int(input())
    if prime(n):
        print('Yes')
    else:
        print('No')

File: 3_1/p012.go
package main

import (
	"fmt"
)

func IsPrime(n int) bool {
	ans := true
	for i := 2; i*i <= n; i++{
		if n % i == 0{
			ans = false
			break
		}
	}
	return ans
}

func main(){
	var n int
	fmt.Scan(&n)
	res := IsPrime(n)
	if res{
		fmt.Println("Yes")
	}else{
		fmt.Println("No")
	}
}
